### DISCUSSION TITLE
Visual revisions

### DISCUSSION THREAD

**Discussion by jameskoster** [WordPress/gutenberg#37037] [Ideas]
It could be really neat to have an option to browse revisions more visually. Something like:

https://user-images.githubusercontent.com/846565/144257944-cfd1472f-bc0f-427b-856d-4a42e33f40ac.mp4

**Comment by shaunandrews** [comment]
Definitely! I've explored this a few time in the past (I can't find the mockups; Still looking...) and have a few thoughts:

- I'd stick to the term "History" as it feels more human.
- Along with document-level history, I think it could be very interesting to explore block-level history.
- - While your video focuses on the Site Editor, I think the Post Editor would benefit from a visual history; Its worth considering if a Post History would require it's own interface.
- Having a way to identify "large" changesets at a quick glance could be very helpful; I recall using varied sized dots and color to help draw attention to major changes in the timeline.
- As you step through the timeline its not always obvious _what_ has changed. Perhaps some highlighting of blocks could help.
- Not all changes to the document might be _visual_ changes; For example, changing a posts category, or changing a page title when the Page Title block is not shown.
- A side-by-side comparison of two changesets could be helpful, especially in the context of a post.
- Remember Apple's Time Machine? (https://support.apple.com/en-us/HT204015) There might be some fun and wacky UI's worth exploring...

**Comment by jameskoster** [comment]
> Not all changes to the document might be visual changes; For example, changing a posts category, or changing a page title when the Page Title block is not shown.

Oh certainly. It probably makes sense for the diff-esque interface to still exist.

> A side-by-side comparison of two changesets could be helpful, especially in the context of a post.

That's a cool idea.

> I think it could be very interesting to explore block-level history

That one kind of blows my mind.

**Comment by shaunandrews** [comment]
>> Not all changes to the document might be visual changes; For example, changing a posts category, or changing a page title when the Page Title block is not shown.

> Oh certainly. It probably makes sense for the diff-esque interface to still exist.

Maybe, but also maybe there are ways to show those non-visual changes in somewhere alongside the visual preview. My first thought is some sort of drawer (like meta-boxes?) that appears to list non-visual changes as you navigate the timeline.

**Comment by talldan** [comment]
This is great, I think it'd really good to see something like this. One of the other areas currently lacking is the interface when restoring an autosave (either one saved locally or to the server), which I would imagine could work similarly.
